node/bft/src/primary.rs

```diff
@@ -822,36 +822,46 @@ impl<N: Network> Primary<N> {
-                    return Ok(());
-                };
+        for transmission_id in batch_header.transmission_ids() {
+            let worker_id = assign_to_worker(*transmission_id, self.num_workers())?;
+            let Some(worker) = self.workers.get(worker_id as usize) else {
+                debug!("Unable to find worker {worker_id}");
+                return Ok(());
+            };
 
-                // If the transmission is a transaction, compute its execution cost.
-                if let (TransmissionID::Transaction(transaction_id, _), Transmission::Transaction(transaction)) =
-                    (transmission_id, transmission)
-                {
-                    // Deserialize the transaction. If the transaction exceeds the maximum size, then return an error.
-                    let transaction = spawn_blocking!({
-                        match transaction {
-                            Data::Object(transaction) => Ok(transaction),
-                            Data::Buffer(bytes) => {
-                                Ok(Transaction::<N>::read_le(&mut bytes.take(N::MAX_TRANSACTION_SIZE as u64))?)
-                            }
-                        }
-                    })?;
+            let Some(transmission) = worker.get_transmission(*transmission_id) else {
+                debug!("Unable to find transmission '{}' in worker '{worker_id}", fmt_id(transmission_id));
+                return Ok(());
+            };
+
+            // If the transmission is a transaction, check that it satisfies the required properties.
 
+            if let (TransmissionID::Transaction(transaction_id, _), Transmission::Transaction(transaction)) =
+                (transmission_id, transmission)
+            {
+                // Deserialize the transaction. If the transaction exceeds the maximum size, then return an error.
+                let transaction = spawn_blocking!({
+                    match transaction {
+                        Data::Object(transaction) => Ok(transaction),
+                        Data::Buffer(bytes) => {
+                            Ok(Transaction::<N>::read_le(&mut bytes.take(N::MAX_TRANSACTION_SIZE as u64))?)
+                        }
+                    }
+                })?;
+
+                // Get the current block height.
+                let block_height = self.ledger.latest_block_height() + 1;
+
+                // If the `ConsensusVersion` is greater than or equal to V5:
+                //   - ensure the transaction doesn't bring the proposal above the spend limit.
+                // Otherwise:
+                //   - ensure that the transaction does not use constructors, the checksum operand, or edition operand.
+                //   - ensure that `program_checksum`s in deployments are `None`.
+                //    TODO: @d0cd this ^ needs to be updated to the correct version once we have pinned it down.
+                if N::CONSENSUS_VERSION(block_height)? >= ConsensusVersion::V5 {
```

Comment:
@d0cd want to remove this checking logic at the snarkOS level - as we agreed snarkVM level is sufficient?